src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -38,56 +37,723 @@ ActorInit Bg_Dy_Yoseizo_InitVars = {
+    &gGreatFairyGivingUpgradeAnim,      // GREATFAIRY_ANIM_GIVING_UPGRADE
+    &gGreatFairySpinLayDownAnim,        // GREATFAIRY_ANIM_SPIN_LAY_DOWN
+    &gGreatFairyLayDownTransitionAnim,  // GREATFAIRY_ANIM_LAY_DOWN_TRANSITION
+    &gGreatFairyLayingDownAnim,         // GREATFAIRY_ANIM_LAYING_DOWN
+    &gGreatFairyShowingItemAnim,        // GREATFAIRY_ANIM_SHOWING_ITEM
+    &gGreatFairyArmsFoldedAnim,         // GREATFAIRY_ANIM_ARMS_FOLDED
+    &gGreatFairyClappingAnim,           // GREATFAIRY_ANIM_CLAPPING
+    &gGreatFairyTeachSpinAttackAnim,    // GREATFAIRY_ANIM_TEACH_SPIN_ATTACK
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0C270.s")
+void BgDyYoseizo_Init(Actor* thisx, PlayState* play) {
+    BgDyYoseizo* this = THIS;
+
+    this->unk2EC = this->actor.world.pos.y + 40.0f;
+    this->actor.focus.pos = this->actor.world.pos;
+
+    SkelAnime_InitFlex(play, &this->skelAnime, &gGreatFairySkel, &gGreatFairyShowingItemAnim, this->jointTable,
+                       this->morphTable, GREAT_FAIRY_LIMB_MAX);
+
+    this->actionFunc = func_80A0BB08;
+    Actor_SetScale(&this->actor, 0.0f);
+    this->eyeIndex = 0;
+    this->mouthIndex = 0;
+    this->blinkTimer = 0;
+    this->unk2F8 = 0;
+    this->actor.home.rot.z = 1;
+    this->unk302 = 0;
+}
+
+void BgDyYoseizo_Destroy(Actor* thisx, PlayState* play) {
+}
+
+// Has no visible effect since no segment set for manually-controllable eye textures
+void BgDyYoseizo_UpdateEyes(BgDyYoseizo* this) {
+    if (this->blinkTimer != 0) {
+        this->blinkTimer--;
+    }
+
+    if (this->blinkTimer == 0) {
+        this->eyeIndex++;
+        if (this->eyeIndex >= 3) {
+            this->eyeIndex = 0;
+            this->blinkTimer = (s32)Rand_ZeroFloat(60.0f) + 20;
+        }
+    }
+}
+
+void BgDyYoseizo_Bob(BgDyYoseizo* this, PlayState* play) {
+    this->actor.shape.yOffset = Math_SinS(play->gameplayFrames * 1000) * 15.0f;
+}
+
+typedef enum GreatFairyEffectTrajectory {
+    /* 0 */ GREAT_FAIRY_EFFECT_TRAJECTORY_RADIANT, // Dispersing particles, in the Great Fairy's signature colour.
+    /* 2 */ GREAT_FAIRY_EFFECT_TRAJECTORY_FAST_RADIANT = 2,      // As above, but initially move 10 times faster.
+    /* 5 */ GREAT_FAIRY_EFFECT_TRAJECTORY_CONVERGE_ON_PLAYER = 5 // Similar to OoT's healing effect, fixed colour.
+} GreatFairyEffectTrajectory;
+
+/**
+ * Spawn particle effects; see `GreatFairyEffectTrajectory` enum for details
+ *
+ * @param trajectoryType use the `GreatFairyEffectTrajectory` enum.
+ * @param count number to spawn.
+ */
+void BgDyYoseizo_SpawnEffects(BgDyYoseizo* this, s16 trajectoryType, s32 count) {
+    static Color_RGB8 sEffectPrimColors[] = {
+        { 255, 235, 220 }, // Magic
+        { 255, 220, 220 }, // Power
+        { 220, 255, 220 }, // Wisdom
+        { 220, 220, 255 }, // Courage
+        { 255, 255, 200 }, // Kindness
+        { 255, 255, 170 }, // GREAT_FAIRY_EFFECT_TRAJECTORY_CONVERGE_ON_PLAYER
```

Comment:
Kindness? Is this suppose to be indexed by `GreatFairyEffectTrajectory`?

Reply:
Not really. `GREAT_FAIRY_EFFECT_TRAJECTORY_RADIANT`, `GREAT_FAIRY_EFFECT_TRAJECTORY_FAST_RADIANT` are changed to the great fairy's type with  `effectType = GREAT_FAIRY_GET_TYPE(&this->actor);`, so these arrays end up being indexed weirdly